### PR TITLE
CLIO-39855: Add ability to toggle thTable columns

### PR DESCRIPTION
**Pull request by craigcarlyle**
* Added ability to toggle column visibility with the `setVisibleColumns()` function in the `thTableDelegate` service
* Removed `th-table-border-bottom` class, now handling border styling in the `thTable` stylesheet
* Updated specs, documentation to include `setVisibleColumns()`
* Added another `thTable` example to demonstrate column toggling

### CODE REVIEW

File: themis_components/thTable/thTableHeader.service.coffee [hunk 1]
```diff
@@ -22,10 +23,11 @@ TableHeader = (options = {}) ->
 
   opposite = ascending: "descending", descending: "ascending"
 
-  return Object.freeze {
+  return {
```
Comment (craigcarlyle) [inline themis_components/thTable/thTableHeader.service.coffee]: This change isn't ideal, but @seanhealy and I agreed it was the best course of action.

File: themis_components/thTable/examples/8 - Toggle Columns/coffee.coffee [hunk 2]
```diff
@@ -0,0 +1,47 @@
+angular.module 'thDemo', ['ThemisComponents']
+  .controller "DemoController", (SimpleTableDelegate, TableHeader, TableSort) ->
```
Comment (PANCAKES) [inline themis_components/thTable/examples/8 - Toggle Columns/coffee.coffee]: You going to catch flack for this? I am not sure if the rule to move these to separate lines was only in the case we went past 100 chars or not. #style-guide

File: themis_components/thTable/thTableDelegate.service.coffee [hunk 3]
```diff
@@ -58,13 +58,21 @@ angular.module 'ThemisComponents'
         currentSortHeader.deactivateSort() if currentSortHeader?
         currentSortHeader = newSortHeader
 
+    setVisibleColumns = (visibleColumns) ->
+      unless visibleColumns.length == headers.length
+        throw new Error "Array length doesn't match amount of columns."
```
Comment (PANCAKES) [inline themis_components/thTable/thTableDelegate.service.coffee]: "Array length does not match the number of columns."

File: themis_components/thTable/thTableDelegate.service.spec.coffee [hunk 4]
```diff
@@ -165,3 +165,29 @@ describe 'ThemisComponents: Service: thTableDelegate', ->
       expectedSortHeader = headers[0]
       delegate.sortData expectedSortHeader
       expect(numCalled).toBe 2
+
+  describe '#setVisibleColumns', ->
+    it 'calls changes the visibility on the second column to "true"', ->
```
Comment (PANCAKES) [inline themis_components/thTable/thTableDelegate.service.spec.coffee]: this is a bit clumsy, how about

```
context "given column 2 having initial visibility of false", ->
  it "changes the visibility of column 2 to true", ->
```

That reads a bit better IMO, thoughts?

File: themis_components/thTable/readme.md [hunk 5]
```diff
@@ -258,6 +258,9 @@ that you can use:
 * `goToPage(pageIndex)`
   * Manually navigate to the specified page.
 
+* `setVisibleColumns([true, true, false])`
+  * Toggle which columns you want to be visible by passing in an array of either
+    `true` or `false` values. Array length must match the column length.
```
Comment (PANCAKES) [inline themis_components/thTable/readme.md]: Array length must match the number of columns.

File: themis_components/thTable/thSimpleTableDelegate.service.coffee [hunk 6]
```diff
@@ -80,16 +84,19 @@ angular.module 'ThemisComponents'
                   .map -> """<td class="th-table-actions-cell"></td>"""
                   .join ''
                   .concat """
-                    <td class="th-table-actions-cell" colspan=#{colspan}>
+                    <td class="th-table-actions-cell"
+                        colspan=#{colspan}
+                        ng-show="thTable.delegate.headers[#{startColumn - 1}].visible">
```
Comment (PANCAKES) [inline themis_components/thTable/thSimpleTableDelegate.service.coffee]: move the the closing ```>``` to the next line? I am not sure if it is in the guide, but having it on th enew line means we would only have one line changed here. #style-guide

File: themis_components/thTable/thSimpleTableDelegate.service.coffee [hunk 7]
```diff
@@ -80,16 +84,19 @@ angular.module 'ThemisComponents'
...
                   """
       return """
-        <tr class="th-table-actions-row th-table-border-bottom"
+        <tr class="th-table-actions-row"
             ng-repeat-end
             ng-mouseover="hover = true"
             ng-mouseleave="hover = false"
-            ng-class="{'th-table-hover-row': hover}">
+            ng-class="{'th-table-hover-row': hover}"
+            ng-show="thTable.delegate.headers[#{startColumn - 1}].visible">
```
Comment (PANCAKES) [inline themis_components/thTable/thSimpleTableDelegate.service.coffee]: move the the closing ```>``` to the next line? I am not sure if it is in the guide, but having it on th enew line means we would only have one line changed here. #style-guide

File: themis_components/thTable/thTableDelegate.service.spec.coffee (hunk 4 above)
Comment (craigcarlyle) [inline themis_components/thTable/thTableDelegate.service.spec.coffee]: Addressed.

File: themis_components/thTable/thTableDelegate.service.coffee (hunk 3 above)
Comment (craigcarlyle) [inline themis_components/thTable/thTableDelegate.service.coffee]: Addressed.

File: themis_components/thTable/thSimpleTableDelegate.service.coffee (hunk 7 above)
Comment (craigcarlyle) [inline themis_components/thTable/thSimpleTableDelegate.service.coffee]: Addressed.

File: themis_components/thTable/thSimpleTableDelegate.service.coffee (hunk 6 above)
Comment (craigcarlyle) [inline themis_components/thTable/thSimpleTableDelegate.service.coffee]: Addressed.

File: themis_components/thTable/readme.md (hunk 5 above)
Comment (craigcarlyle) [inline themis_components/thTable/readme.md]: Addressed.

File: themis_components/thTable/examples/8 - Toggle Columns/coffee.coffee (hunk 2 above)
Comment (craigcarlyle) [inline themis_components/thTable/examples/8 - Toggle Columns/coffee.coffee]: Addressed.

File: themis_components/thTable/examples/8 - Toggle Columns/coffee.coffee [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+          sortField: 'email'
+      ]
+
+      fetchData: ({sortHeader}, updateData) ->
+        sortedData = sort data, sortHeader
+        updateData {data: sortedData, totalItems: sortedData.length}
+
+    @visibleColumns = [true, true, true]
+
+    @setColumns = (columnNumber) ->
```
Comment (dtomalty) [inline themis_components/thTable/examples/8 - Toggle Columns/coffee.coffee]: Can we change this to `toggleColumn`?

**Pull request merged**